src/hooks/__tests__/useSourceApi.test.ts

```diff
@@ -0,0 +1,310 @@
+        }
+      },
+      [
+        { id: 123, name: 'Lorem' },
+        { id: 456, name: 'Ipsum' }
+      ]
+    );
+
+    // warning
+    callbackSuccess(
+      {
+        data: {
+          deleted: [456],
+          skipped: [{ source: 123 }]
+        }
+      },
+      [
+        { id: 123, name: 'Lorem' },
+        { id: 456, name: 'Ipsum' }
+      ]
+    );
+
+    // success
+    callbackSuccess(
+      {
+        data: {
+          deleted: [123, 456]
+        }
+      },
+      [
+        { id: 123, name: 'Lorem' },
+        { id: 456, name: 'Ipsum' }
+      ]
+    );
+
+    expect(mockOnAddAlert.mock.calls).toMatchSnapshot('callbackSuccess');
+  });
+
+  it('should process an API error response', () => {
+    const { callbackError } = hookResult;
+
+    callbackError(
+      {
+        response: {
+          data: {
+            detail: 'Lorem ipsum'
+          }
+        }
+      },
+      [
+        { id: 123, name: 'Lorem' },
+        { id: 456, name: 'Ipsum' }
+      ]
+    );
+
+    callbackError(
+      {
+        response: {
+          data: {
+            message: 'Dolor sit'
+          }
+        }
+      },
+      [
+        { id: 123, name: 'Lorem' },
+        { id: 456, name: 'Ipsum' }
+      ]
+    );
+
+    callbackError(
+      {
+        message: 'Amet'
+      },
+      [
+        { id: 123, name: 'Lorem' },
+        { id: 456, name: 'Ipsum' }
+      ]
+    );
+
+    callbackError({}, [
+      { id: 123, name: 'Lorem' },
+      { id: 456, name: 'Ipsum' }
+    ]);
+
+    expect(mockOnAddAlert.mock.calls).toMatchSnapshot('callbackError');
+  });
+});
+
+describe('useAddSourceApi', () => {
+  let mockOnAddAlert;
+  let hookResult;
+
+  beforeEach(() => {
+    mockOnAddAlert = jest.fn();
+    const hook = renderHook(() => useAddSourceApi(mockOnAddAlert));
+    hookResult = hook?.result?.current;
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('should attempt an api call to add sources', () => {
+    const { apiCall } = hookResult;
+    const spyAxios = jest.spyOn(axios, 'post');
+
+    apiCall({ name: 'Lorem' });
+    expect(spyAxios.mock.calls).toMatchSnapshot('apiCall');
+  });
+
+  it('should handle success while attempting to add a source', async () => {
+    const { addSources } = hookResult;
+    jest.spyOn(axios, 'post').mockImplementation(() =>
+      Promise.resolve({
+        data: { name: 'Lorem', id: '123' }
+      })
+    );
+
+    await addSources({ name: 'Lorem' });
+    expect(mockOnAddAlert.mock.calls).toMatchSnapshot('addSources, success');
+  });
+
+  it('should handle errors while attempting to add a source', async () => {
+    const { addSources } = hookResult;
+    jest.spyOn(axios, 'post').mockImplementation(() =>
+      Promise.reject(
+        Object.assign(new Error('Mock error'), {
+          isAxiosError: true,
+          response: {
+            data: {
+              message: 'Mock error'
+            }
+          }
+        })
+      )
+    );
```

Comment:
required: same comment before about over-engineering our tests, remove the `Object.assign` in favor of a simplistic object